rectools/models/sasrec.py

```diff
@@ -618,10 +610,12 @@ def _fit(
         train_dataloader = self.data_preparator.get_dataloader_train(processed_dataset)
 
         self.model = deepcopy(self._model)  # TODO: check that it works
-        self.model.costruct_item_net(processed_dataset)
+        self.model.construct_item_net(processed_dataset)
+        self.lightning_model = SasRec(self.model, self.lr, self.loss)
+
+        self.trainer = deepcopy(self._trainer)
 
-        self.trainer.fit(self.model, train_dataloader)
-        self.model = self.trainer.model
+        self.trainer.fit(self.lightning_model, train_dataloader)
```

Comment:
Did you check if `self.model` weights are updated inplace after this?
In recommend you are using `self.lightning_model.torch_model` not `self.model`.
I think it's better to use pure torch net there.